pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -45,10 +45,8 @@ type Reconciler struct {
 	kubeClientSet kubernetes.Interface
 	// operatorClientSet allows us to configure operator objects
 	operatorClientSet clientset.Interface
-	// config is the manifest of KnativeServing
-	config mf.Manifest
-	// targetVersion is the version of the KnativeServing manifest to install
-	targetVersion string
+	// manifest is empty but with a valid client and logger
```

Comment:
maybe Manifest is initialized empty?

Reply:
There is a tricky point that knativeserving/knativeeventing reconciler and controller do not share the same context, but we need the correct context in controller to create the correct `client and logger`.
Initially, we thought of creating manifest in reconciler, but it seemed we have to leverage the context in controller to create client and logger.
We are only able to know which version to be installed(target) in reconciler, by spec.version.
As you see, manifest is prepared in controller, but filled in resources in reconciler.

Reply:
I mostly meant // manifest is empty 
implied to me that it stays empty, whereas really it just *starts* empty.

Reply:
It does actually stay empty. Each time a reconcile appends to it, that creates a brand new, deep-copy of it -- recall all manifests are immutable.

Reply:
Ah! Then the comment is useful :) I did find that a bit surprising.

It might be better to be a bit more explicit and construct the Manifest below, but I don't feel too strongly about it so long as the comment stays. Maybe we could add the word "immutable" here as well?

Reply:
Done